mautic/mautic
# Disable tracking 404

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | fixes https://github.com/mautic/mautic/issues/7565
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
Based on discussion https://github.com/mautic/mautic/issues/7565 and comment https://github.com/mautic/mautic/issues/7565#issuecomment-497675707 
we've added option Do Not Track 404 error page tracking for anonymous visitors. This causes a lot of anonymous contacts from bots. New option added to Configuration > Tracking Settings tab.  
UNIT test included in this PR

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Try in incognito not existed url: https://yourmautic.com/whatever
2. Go to Mautic > Contacts and switch to anonymous list view
3. See new contact was create with https://yourmautic.com/whatever page hit

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Repeat all steps but disable in Configuration > Tracking Settings tab > Do Not Track 404 error page
3. See If 404 not found page wasn't tracked
4. Create identified contact (via form, via tracking code)
5. Go to 404 not found page and see If page was tracked

![image](https://user-images.githubusercontent.com/462477/132746748-04d4edae-1be9-40c9-b3b4-aded0d0ee023.png)


<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/7614"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
